stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -1620,6 +1628,54 @@ logdet(A) = log(det(A))
 
 const NumberArray{T<:Number} = AbstractArray{T}
 
+exactdiv(a, b) = a/b
+exactdiv(a::Integer, b::Integer) = div(a, b)
+
+"""
+    LinearAlgebra.det_bareiss!(M)
+
+Calculates the determinant of a matrix using the
+[Bareiss Algorithm](https://en.wikipedia.org/wiki/Bareiss_algorithm) using
+inplace operations.
+
+# Examples
+```jldoctest
+julia> M = [1 0; 2 2]
+2×2 Matrix{Int64}:
+ 1  0
+ 2  2
+
+julia> LinearAlgebra.det_bareiss!(M)
+2
+```
+"""
+function det_bareiss!(M)
```

Comment:
Is there a reason to not just call this function `det!(A::AbstractMatrix{BigInt})` I'm not sure if giving it it's own name makes things clearer.

Reply:
> My suggestion would be to have a non-exported `LinearAlgebra.det_bareiss` function that accepts any element type, and calls something like `exactdiv(a,b)=a/b; exactdiv(a::Integer,b::Integer) = div(a,b)`.  

>Then define only `det(A::AbstractMatrix{BigInt}) = det_bareiss(A)`, but packages that define other ring types can define additional `det` methods as appropriate or call `det_bareiss` directly.

Is the line of reasoning that @stevengj  had, as far as the code is concerned, I have no problem doing it either way.
Thanks for the review!